elephant/spectral.py

```diff
@@ -242,6 +242,129 @@ def welch_psd(signal, n_segments=8, len_segment=None,
     return freqs, psd
 
 
+def multitaper_psd(signal, fs=1, nw=4, num_tapers=None,
+                   peak_resolution=None):
+    """
+    Estimates power spectrum density (PSD) of a given 'neo.AnalogSignal'
+    using Multitaper method
+
+    The PSD is obtained through the following steps:
+
+    1. Calculate 'num_tapers' approximately independent estimates of the
+       spectrum by multiplying the signal with the discrete prolate spheroidal
+       functions (also known as Slepian function) and calculate the PSD of the
+       products
+
+    2. Average the approximately independent estimates to decrease overall
+       variance of the estimates
+
+    Parameters
+    ----------
+    signal : neo.AnalogSignal
+        Time series data of which PSD is estimated. When `signal` is np.ndarray
+        sampling frequency should be given through keyword argument `fs`.
+    fs : float, optional
+        Specifies the sampling frequency of the input time series
+        Default: 1.0
+    nw : float, optional
+        Time bandwidth product
+        Default: 4.0
+    num_tapers : int, optional
+        Number of tapers used in 1. to obtain estimate of PSD. By default
+        [2*nw] - 1 is chosen.
+        Default: None
+    peak_resolution : float, optional
+        Desired frequency resolution of the obtained PSD estimate. When given
+        as a `float`, it is taken as frequency in Hz.
+        Default: None.
+
+    Notes
+    -----
+    There is a parameter hierarchy regarding nw, num_tapers and
+    peak_resolution. If peak_resolution is provided, it determines both nw and
+    the num_tapers. Specifying num_tapers has an effect only if peak_resolution
+    is not provided.
```

Comment:
This needs to be written in a way that explains what `peak_resolution` is and what it does to a person who does not know the method. Currently, it does not really explain what `peak_resolution` is.